src/backend/marsha/core/tests/test_api_video.py

```diff
@@ -1996,6 +2012,24 @@ def test_api_video_patch_detail_token_user_description(self):
         video.refresh_from_db()
         self.assertEqual(video.description, "my new description")
 
+    def test_api_video_patch_detail_token_user_is_public(self):
+        """Token users should be able to patch the public flag of their video through the API."""
+        video = factories.VideoFactory(is_public=False)
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        data = {"is_public": True}
+        response = self.client.patch(
+            f"/api/videos/{video.id}/",
+            data,
+            HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        video.refresh_from_db()
+        self.assertTrue(video.is_public)
+
```

Comment:
I assume there is a test to confirm that put/patch is not allowed for students or anonymous users. Should we make a special one for this sensitive field or you think it is already safe?

Reply:
I see no test for `student` 😱

Reply:
I added tests for post, put, patch and delete requests.